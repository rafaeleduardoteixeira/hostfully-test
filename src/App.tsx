import React from 'react';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import createdRoutes from './constants/router';
import { Header } from './components/Header/Header';
import { AppContainer } from './App.styles';

function App() {
  return (
    <AppContainer>
      <Header />
      <RouterProvider router={createdRoutes} />
    </AppContainer>
  );
}

export default App;
