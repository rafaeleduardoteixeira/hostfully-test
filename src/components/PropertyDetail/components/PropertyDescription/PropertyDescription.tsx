import { Property } from '../../../../interfaces/property';
import {
  LineBottom,
  PropertyDescriptionContainer,
  PropertyDescriptionText,
  PropertyDescriptionTitle,
} from './PropertyDescription.styles';

interface PropertyDescriptionProps {
  property: Property;
}

export const PropertyDescription = ({ property }: PropertyDescriptionProps) => {
  return (
    <PropertyDescriptionContainer>
      <PropertyDescriptionTitle>Description</PropertyDescriptionTitle>
      <LineBottom />
      <PropertyDescriptionText>{property.description}</PropertyDescriptionText>
    </PropertyDescriptionContainer>
  );
};
