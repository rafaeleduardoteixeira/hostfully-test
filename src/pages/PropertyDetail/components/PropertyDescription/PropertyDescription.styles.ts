import styled from 'styled-components';
import { colors } from '../../../../themes';

export const PropertyDescriptionContainer = styled.div`
  margin-top: 20px;
`;

export const PropertyDescriptionTitle = styled.h2`
  color: ${colors.hostfullySecondaryColor};
  font-size: 24px;
  font-weight: 700;
`;

export const LineBottom = styled.div`
  border-bottom: 2px solid ${colors.hostfullyPrimaryColor};
  width: 300px;
  margin: 5px 0px 20px 0px;
`;

export const PropertyDescriptionText = styled.p`
  font-size: 16px;
  font-weight: 400;
  color: ${colors.grayDark};
`;
