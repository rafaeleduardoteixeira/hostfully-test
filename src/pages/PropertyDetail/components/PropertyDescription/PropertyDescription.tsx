import React from 'react';
import { Property } from '../../../../interfaces/property';
import {
  LineBottom,
  PropertyDescriptionContainer,
  PropertyDescriptionText,
  PropertyDescriptionTitle,
} from './PropertyDescription.styles';

interface PropertyDescriptionProps {
  property: Property;
}

export const PropertyDescription = ({ property }: PropertyDescriptionProps) => {
  return (
    <PropertyDescriptionContainer data-testid="PropertyDescription">
      <PropertyDescriptionTitle>Description</PropertyDescriptionTitle>
      <LineBottom />
      <PropertyDescriptionText>{property.description}</PropertyDescriptionText>
    </PropertyDescriptionContainer>
  );
};
