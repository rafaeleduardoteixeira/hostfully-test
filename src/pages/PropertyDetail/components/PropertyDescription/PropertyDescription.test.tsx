import React, { act } from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { PropertyDescription } from './PropertyDescription';
import { AppContext } from '../../../../Context/App.context';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('react-leaflet', () => jest.fn());

describe('PropertyDescription', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(
      <AppContext.Provider
        value={{
          isLoading: false,
          getProperties: jest.fn(),
          getPropertyById: jest.fn(),
          getPropertyRentedDates: jest.fn(),
          getRentedProperties: jest.fn(),
          handleAddRentals: jest.fn(),
          handleRemoveRentals: jest.fn(),
          handleEditRentals: jest.fn(),
        }}
      >
        <PropertyDescription
          property={{
            id: '1',
            name: 'Name 1',
            description: 'Description 1',
            city: 'City 1',
            country: 'Country 1',
            purpose: 'Purpose 1',
            price: 1,
            images: ['image'],
            position: {
              latitude: 0,
              longitude: 0,
            },
          }}
        />
      </AppContext.Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should render PropertyDescription', () => {
    render(
      <AppContext.Provider
        value={{
          isLoading: false,
          getProperties: jest.fn(),
          getPropertyById: jest.fn(),
          getPropertyRentedDates: jest.fn(),
          getRentedProperties: jest.fn(),
          handleAddRentals: jest.fn(),
          handleRemoveRentals: jest.fn(),
          handleEditRentals: jest.fn(),
        }}
      >
        <PropertyDescription
          property={{
            id: '1',
            name: 'Name 1',
            description: 'Description 1',
            city: 'City 1',
            country: 'Country 1',
            purpose: 'Purpose 1',
            price: 1,
            images: ['image'],
            position: {
              latitude: 0,
              longitude: 0,
            },
          }}
        />
      </AppContext.Provider>,
    );

    const PropertyDescriptionElement = screen.getByTestId('PropertyDescription');
    expect(PropertyDescriptionElement).toBeInTheDocument();
  });
});
