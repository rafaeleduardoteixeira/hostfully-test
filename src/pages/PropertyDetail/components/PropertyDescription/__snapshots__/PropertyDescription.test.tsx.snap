// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDescription Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu dAccIH"
    data-testid="PropertyDescription"
  >
    <h2
      class="sc-gJhJTp dMqlJz"
    >
      Description
    </h2>
    <div
      class="sc-elDIKY keOpyt"
    />
    <p
      class="sc-fQpRED hHDnMH"
    >
      Description 1
    </p>
  </div>
</DocumentFragment>
`;
